### PR TITLE
chore:  apply refactor encryption at rest

### PR DESCRIPTION
**Pull request by netbe**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Applying commit "refactor: encryption at rest FS-1596 (#211)" for callkit release

* Adapt some methods for mls
----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift [hunk 1]
```diff
@@ -72,8 +72,11 @@ extension EventDecoder {
     ///   - events: Encrypted events
     ///   - block: A block that receives the decrypted events for processing.
 
+    ///   - block: A block that receives the decrypted events for processing.
+
```
Comment (David-Henner) [inline wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift:76]:
```suggestion
```

File: wire-ios-sync-engine/Tests/Source/UserSession/OperationStatusTests.swift [hunk 2]
```diff
@@ -52,15 +52,6 @@ class OperationStatusTests: MessagingTest {
...
 
-    func testOperationState_whenInBackgroundWithOngoingCall() {
-        // when
-        sut.isInBackground = true
-        sut.hasOngoingCall = true
-
-        // then
-        XCTAssertEqual(sut.operationState, .backgroundCall)
-    }
-
```
Comment (johnxnguyen) [inline wire-ios-sync-engine/Tests/Source/UserSession/OperationStatusTests.swift]: @David-Henner I think this wasn't supposed to be deleted

File: wire-ios-system/Source/Logging/Flow.swift
Comment (johnxnguyen) [inline wire-ios-system/Source/Logging/Flow.swift:1]: This wasn't part of the refactoring but I guess it doesn't hurt to have it.

File: wire-ios-system/Source/Logging/WireLogger.swift
Comment (johnxnguyen) [inline wire-ios-system/Source/Logging/WireLogger.swift:1]: Also most of these changes weren't from the refactoring, but it's good to have them here anyway.

Comment (netbe) [reply wire-ios-system/Source/Logging/WireLogger.swift:1]: yes it was easier to have it

File: wire-ios-system/Source/Logging/Flow.swift
Comment (netbe) [reply wire-ios-system/Source/Logging/Flow.swift:1]: same I just reapply the commit

File: wire-ios-sync-engine/Tests/Source/UserSession/OperationStatusTests.swift (hunk 2 above)
Comment (David-Henner) [reply wire-ios-sync-engine/Tests/Source/UserSession/OperationStatusTests.swift]: correct

**Pull request merged**